WFLY-14316 Replace 'Interceptors' references in java doc with corresp…

RanabirChakraborty opened the pull request:
Issue: https://issues.redhat.com/browse/WFLY-14316
(Replace "Interceptors" references in java doc with corresponding Jakarta EE ones)
Link with: https://github.com/wildfly/wildfly/pull/13808

RanabirChakraborty committed:
WFLY-14316 Replace 'Interceptors' references in java doc with corresponding Jakarta EE ones

RanabirChakraborty commented:
@yersan if you can review this :)

CC: @asoldano @scottmarlow

asoldano commented on docs/src/main/asciidoc/Getting_Started_Guide.adoc:
```diff
@@ -64,7 +64,7 @@ Profile |WildFly {wildflyVersion} Full Platform |WildFly {wildflyVersion} Web Pr
 2.0Not Available |X(Lite)
 
-|JSR-318: Interceptors 1.2 |X |X |X |X
+|JSR-318: Jakarta Interceptors 1.2 |X |X |X |X
```
I think the reference to JSR-318 should be removed here

yersan replied:
+1

asoldano commented on docs/src/main/asciidoc/_developer-guide/Jakarta_XML_Web_Services/Apache_CXF_integration.adoc:
```diff
@@ -427,8 +427,8 @@ Schema validation of exchanged messages can also be enabled in
 
-[[interceptors]]
-==== Interceptors
+[[Jakarta_Interceptors]]
+==== Jakarta Interceptors
```
I believe this should not be changed, as the interceptors discussed here are Apache CXF interceptors, not JavaEE/JakartaEE interceptors. @jimma , please confirm.

jimma replied:
Here is CXF interceptor and we don't need to change it. @asoldano @RanabirChakraborty @yersan

asoldano commented on docs/src/main/asciidoc/_developer-guide/Jakarta_XML_Web_Services/Apache_CXF_integration.adoc:
```diff
@@ -670,3 +670,5 @@ disabling CN host name check or not
 
 The vanilla Apache CXF defaults apply when the system properties above
 are not set.
+
+NOTE: References in this document to Interceptors refer to the Jakarta Interceptors unless otherwise noted.
```
@RanabirChakraborty , @jimma , please double-check this as well, so that CXF interceptors are not confused with JakartaEE interceptors.

jimma replied:
@RanabirChakraborty  Again, here is cxf interceptor.

asoldano commented on docs/src/main/asciidoc/_developer-guide/Jakarta_XML_Web_Services/WS-Security.adoc:
```diff
@@ -1169,3 +1169,5 @@ public class ServiceImpl implements ServiceIface {
    ...
 }
 ----
+
+NOTE: References in this document to Interceptors refer to the Jakarta Interceptors unless otherwise noted.
```
Same as above, I believe CXF interceptors are referenced here. (/cc @jimma )

jimma replied:
Yes . CXF interceptor.

asoldano commented on ee/src/main/java/org/jboss/as/ee/component/ProxyInvocationHandler.java:
```diff
@@ -48,7 +48,7 @@
      * Construct a new instance.
      *
-     * @param interceptors the interceptors map to use
+     * @param Jakarta Interceptors the Jakarta Interceptors map to use
```
Please be aware of the javadoc syntax; the `interceptors` just after `@param` can't be changes as it must match the actual parameter name in the method that follows this javadoc block.

RanabirChakraborty replied:
That was a mistake! I'll change it :)

asoldano commented on security/api/src/main/java/org/jboss/as/security/api/ConnectionSecurityContext.java:
```diff
@@ -45,7 +45,7 @@
  * As a connection is established to the application server the remote user is authenticated, this API allows the
  * {@link Collection} of {@link Principal}s for the remote user to be obtained, the API then allows for an alternative identity
- * to be pushed by interceptors for validation in the security interceptors for subsequent EJB invocations.
+ * to be pushed by Jakarta Interceptors for validation in the security Jakarta Interceptors for subsequent EJB invocations.
```
I suspect the interceptors here are something different from the JakartaEE interceptors

asoldano reviewed (commented):
Hi @RanabirChakraborty , besides for the comments I provided, I generally think we should not apply any change to "interceptors" references unless there's a very clear reference to JavaEE / EJB / CDI  "interceptors".

yersan commented on ee/src/main/java/org/jboss/as/ee/component/InterceptorEnvironment.java:
```diff
@@ -26,7 +26,7 @@
 
 /**
- * Class that represents the environment entries of an interceptor, defined in the interceptors section
+ * Class that represents the environment entries of an interceptor, defined in the Jakarta Interceptors section
```
This comment is making a reference to a section on the ejb-jar.xml, I've my doubts about whether should be changed, @asoldano Do you have any opinion about it?

yersan replied:
For clarity, I just suggest replacing it with " ... defined in the <interceptors> section ..."

asoldano replied:
if the section is named "interceptors", we would leave this as is, I think..

yersan reviewed (changes requested):
I think this one should be changed too, check EJB3_Clustered_Database_Timers.adoc:123

s/Implement an EE interceptor that enables/Jakarta EE interceptor that enables

RanabirChakraborty commented:
@asoldano @yersan @jimma Thanks for the review! Did the changes accordingly :)

yersan reviewed (changes requested):
@RanabirChakraborty I think you could have missed this comment https://github.com/wildfly/wildfly/pull/14057#pullrequestreview-600584413

Could you take a look, indeed, there are some "EE interceptor" references in this file. I think we could either use `Jakarta Interceptor` or just `interceptor`.

In general, we are using either `Jakarta Interceptors` or `Jakarta EE Interceptors`. Both are considered valid ones, but please take a look at `Jakarta EE interceptors` references, they should be `Jakarta EE Interceptors`. 

In addition, could you also remove the line break at the end of Container_interceptors.adoc:174? A line break in the middle of a title makes it render badly in the final documentation. 
The title should be "Semantic difference between container interceptor(s) and Jakarta EE Interceptor(s) API", and all in just one line.

RanabirChakraborty commented on weld/common/src/main/java/org/jboss/as/weld/interceptors/Jsr299BindingsInterceptor.java:
```diff
@@ -41,10 +41,10 @@
 
 /**
- * Interceptor for applying the JSR-299 specific interceptor bindings.
+ * Jakarta Interceptors for applying the JSR-299 specific Jakarta Interceptors bindings.
```
yeah true! I'll remove JSR-299 reference.

yersan replied:
Good question, I think we can just use the generic one and then replace the sentence with "Jakarta Interceptors for applying Interceptors bindings"

yersan commented on weld/common/src/main/java/org/jboss/as/weld/interceptors/Jsr299BindingsInterceptor.java:
```diff
@@ -41,10 +41,10 @@
  * <p/>
- * It is a separate interceptor, as it needs to be applied after all
- * the other existing interceptors.
+ * It is a separate Jakarta Interceptors, as it needs to be applied after all
+ * the other existing Jakarta Interceptors.
```
Leave this as they were before:

It is a separate interceptor, as it needs to be applied after all the other existing interceptors.

yersan reviewed (changes requested):
I still don't know what you are going to do with `EJB3_Clustered_Database_Timers.adoc` and the `EE interceptor` references on that file.

There is also another reference in testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/container/interceptor/FlowTrackingBean.java:35

In addition, could you also check the upper case on all the "Jakarta EE interceptors", they should be either "Jakarta Interceptors" or "Jakarta EE Interceptors"

RanabirChakraborty commented:
@yersan Thanks! 
because of the future conflicts, I have changed the same here - https://github.com/wildfly/wildfly/pull/14068/files
EJB3_Clustered_Database_Timers.adoc - [here](https://github.com/wildfly/wildfly/blob/11519029ef51e1b24bb20b702c83553b951ec965/docs/src/main/asciidoc/_developer-guide/ejb3/Jakarta_Enterprise_Beans_3_Clustered_Database_Timers.adoc)
Container_interceptors.adoc - [here](https://github.com/wildfly/wildfly/blob/11519029ef51e1b24bb20b702c83553b951ec965/docs/src/main/asciidoc/_developer-guide/ejb3/Container_interceptors.adoc)
FlowTrackingBean.java - [here](https://github.com/wildfly/wildfly/blob/11519029ef51e1b24bb20b702c83553b951ec965/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/container/interceptor/FlowTrackingBean.java)
ContainerInterceptorsTestCase.java - [here](https://github.com/wildfly/wildfly/blob/11519029ef51e1b24bb20b702c83553b951ec965/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/container/interceptor/ContainerInterceptorsTestCase.java)

and I have changed `Jakarta EE Interceptors` and `EE Interceptors` both to `Jakarta Interceptors` according to https://jakarta.ee/specifications/interceptors/

yersan commented on weld/common/src/main/java/org/jboss/as/weld/interceptors/Jsr299BindingsInterceptor.java:
```diff
@@ -41,9 +41,9 @@
+ * Jakarta Interceptors for applying interceptors bindings.
  * <p/>
- * It is a separate interceptor, as it needs to be applied after all
+ * It is a separate interceptors, as it needs to be applied after all
```
s/It is a separate interceptors/It is a separate interceptor

RanabirChakraborty replied:
Thanks! Done :)

yersan reviewed (approved):
Looks fine, thanks @RanabirChakraborty
